src/components/TodoModal/TodoModal.tsx

```diff
@@ -15,25 +32,35 @@
               className="modal-card-title has-text-weight-medium"
               data-cy="modal-header"
             >
```

Comment:
You're generating a new string on each render by using a template literal for the `Todo #${activeTodo.id}` text. It's better to avoid generating new values during render because it can lead to unnecessary re-renders and negatively impact performance. Consider moving this value outside of the component or memoizing it with `useMemo`.